### PR TITLE
preserve data locally

### PR DESCRIPTION
**Pull request by Ursaaaaaa**
- Add local storage
- Added variables for functions
- Added function to save input locally on submit with submit event listener
- Added function to also save input locally with input event listener

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -153,4 +155,17 @@ form.addEventListener('submit', (e) => {
...
+  };
+  localStorage.setItem('msg', JSON.stringify(msg));
+});
+
+if (localStorage.getItem('msg') != null) {
+  const userMsg = JSON.parse(localStorage.getItem('msg'));
+  inputEm.value = userMsg.email;
+  inputNm.value = userMsg.name;
+  inputMs.value = userMsg.msg;
+}
```
Comment (Mre55) [inline index.js]: Nice job 👍 persisting the data to local storage and getting it back on page load. But as required in the [project description](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md#project-requirements) the form data should have to be saved when the user changes the content of any input field, the data is saved to the local storage. Currently, the data will be saved when the user submits the form. Kindly, add a function to save the form data to local storage after the user leaves that particular field. For example, after entering the name and when typing the email the data should be saved in the local storage the same for email and for the message.

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @ramrod433,**

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The README is well documented ✅ 
- Linters are working properly ✅ 

## Required Changes ♻️🛠️ 

- [x] Please add a short summary to the pull request description to let know the code reviewer what you have done in this project. Currently, it's vague and generic.

_Check the comments under the review._



## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.js (hunk 1 above)
Comment (Ursaaaaaa) [reply index.js]: Thanks. Changes made

**Review by Lordkaito — APPROVED** [review]
Hi @ramrod433, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**